### PR TITLE
[AKS] add support for creation time node labels

### PR DESCRIPTION
**Pull request by xizhamsft**
**History Notes:**  

[AKS] add --nodepool-labels when creating azure kubernetes cluster
[AKS] add --labels when adding a new nodepool to azure kubernetes cluster

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by jluk** [comment]
Adding context here that we would like to have this join the 2.1.1 version release on [S166](https://github.com/Azure/azure-cli/milestone/83). We will subsequently be trying to merge this exact change into the cli-extension for aks preview customers as well.

This will close the final stage of this long awaited AKS feature: https://github.com/Azure/AKS/issues/1088

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -113,6 +113,8 @@ az webapp create : Fixing issue when running the command with --runtime
 * Each cluster gets a separate service principal to improve isolation
 * Add --nodepool-tags to node pool when creating azure kubernetes cluster
 * Add --tags when adding or updating a nodepool to cluster
+* Add --nodepool-labels when creating azure kubernetes cluster
+* Add --labels when adding a nodepool to cluster
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst]: Please remove the history notes here, it's in a previous version 2.0.81. You've already added it in the PR description. Our release script will fetch the history notes in PR title or PR description and insert into HISTORY.rst before release. You no longer need to modify HISTORY.rst in a regular PR.

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 2]
```diff
@@ -259,3 +259,75 @@ def validate_vnet_subnet_id(namespace):
...
+    else:
+        after_validation_labels = validate_label(labels)
+
+    if hasattr(namespace, 'nodepool_labels'):
+        namespace.nodepool_labels = after_validation_labels
+    else:
+        namespace.labels = after_validation_labels
+
+
+def validate_label(label):
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py]: Maybe you can add there rules to help message.

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 3]
```diff
@@ -259,3 +259,75 @@ def validate_vnet_subnet_id(namespace):
...
+    """Validates that provided label is a valid format"""
+    prefix_regex = re.compile(r"^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$")
+    name_regex = re.compile(r"^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$")
+    value_regex = re.compile(r"^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$")
+
+    if label == "":
+        return {}
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py]: Do you want to support clearing labels with `--labels ""`? Just like `tags`?

Comment (xizhamsft) [reply src/azure-cli/azure/cli/command_modules/acs/_validators.py]: We will in the future. But for short to mid term, we don't allow labels update after creation.

File: src/azure-cli/HISTORY.rst (hunk 1 above)
Comment (xizhamsft) [reply src/azure-cli/HISTORY.rst]: Nice. Good to know!

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py (hunk 2 above)
Comment (xizhamsft) [reply src/azure-cli/azure/cli/command_modules/acs/_validators.py]: The syntax for labels are too long. I included a link to the definition: https://aka.ms/node-labels in the help. That should help.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/acs/_validators.py]: Thank you!

**Pull request merged**